requestmatch/matcher.go

```diff
@@ -140,8 +159,8 @@ func makeLeaf(d Definition) (*leafMatcher, error) {
 		method:         d.Method(),
 		hostRxs:        hostRxs,
 		pathRxs:        pathRxs,
-		headersExact:   d.Headers(),
-		headersRegexps: allHeaderRxs,
+		headersExact:   keyValToLower(d.Headers()),
+		headersRegexps: headerRegexpsToLower(allHeaderRxs),
```

Comment:
Why did you decide to go all lowercase on the header instead of the more common canonicalised form?

The godoc says:

        // HTTP defines that header names are case-insensitive.
        // The request parser implements this by canonicalizing the
        // name, making the first character and any characters
        // following a hyphen uppercase and the rest lowercase.

And uses `func CanonicalHeaderKey(s string) string` for that